infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -292,19 +321,30 @@ public void multiRead(CorfuPayloadMsg<MultipleReadRequest> msg, ChannelHandlerCo
         boolean cacheable = msg.getPayload().isCacheReadResult();
         log.trace("multiRead: {}, cacheable: {}", msg.getPayload().getAddresses(), cacheable);
 
-        ReadResponse rr = new ReadResponse();
+        Lock lock = resetLock.readLock();
+        lock.lock();
+
         try {
-            for (Long address : msg.getPayload().getAddresses()) {
-                ILogData logData = dataCache.get(address, cacheable);
-                if (logData == null) {
-                    rr.put(address, LogData.getEmpty(address));
-                } else {
-                    rr.put(address, (LogData) logData);
+            ReadResponse rr = new ReadResponse();
+            try {
+                if (msg.getEpoch() != serverContext.getServerEpoch()) {
+                    throw new WrongEpochException(serverContext.getServerEpoch());
```

Comment:
You have to send back a `WRONG_EPOCH` and not throw an exception. Otherwise, the exception wont be handled.

Reply:
fixed